django/db/models/base.py

```diff
@@ -1655,13 +1657,29 @@ def _check_long_column_names(cls):
 
         return errors
 
+    @classmethod
+    def _check_constraints(cls):
+        errors = []
+        if connection.features.supports_table_check_constraints:
```

Comment:
I was confused by this briefly, I think you should change the below to be

```
if (
    not connection.features.supports_table_check_constraints and
    any(isinst...
):
    errors.append(
```

Reply:
This should use the same logic in `_check_long_column_names()` rather than only examining the default database connection.